Core/clim-basic/extended-streams/recording.lisp

```diff
@@ -1622,17 +1625,22 @@ were added."
                          align-y toward-x toward-y transform-glyphs))))
 
 (defrecord-predicate draw-text-output-record
-    (string start end point-x point-y align-x align-y toward-x toward-y transform-glyphs)
-  (and (if-supplied (string)
-         (string= (slot-value record 'string) string))
-       (if-supplied (start)
-         (eql (slot-value record 'start) start))
-       (if-supplied (end)
-         (eql (slot-value record 'end) end))
-       (if-supplied (point-x coordinate)
+    (string (:start start nil) (:end end nil)
+     point-x point-y align-x align-y toward-x toward-y transform-glyphs)
+  ;; Compare position first because it is cheap and an update is most
+  ;; likely to change the position.
+  (and (if-supplied (point-x coordinate)
          (coordinate= (slot-value record 'point-x) point-x))
        (if-supplied (point-y coordinate)
          (coordinate= (slot-value record 'point-y) point-y))
+       (if-supplied (string)
+         (string= (slot-value record 'string) string)
+         (let ((start2 0)
+               (end2   nil))
+           (if-supplied (start) (setf start2 start))
+           (if-supplied (end) (setf end2 end))
+           (string= (slot-value record 'string) string
+                    :start2 start2 :end2 end2)))
```

Comment:
ditto, no need for separate forms for existing and non-existing slots.

Reply:
of course, that requires somewhat intimate knowledge of the implementer what is a slot and what is not, but that requirement is true for both cases.